### PR TITLE
fix #38418 imaginetricks.com

### LINKED ISSUE #38418: imaginetricks.com
### Issue URL (Anti Adblock Script)
[https://www.imaginetricks.com/bypass-sms-verification-via-disposable-mobile-number-for-verification/](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww.imaginetricks.com%2Fbypass-sms-verification-via-disposable-mobile-number-for-verification%2F)

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>

### System configuration

Information | value
--- | ---
Platform: | Ext
AdGuard version: | 3.2.1
Browser: | Chrome
Filters: | EasyList China,<br>AdGuard Annoyances,<br>AdGuard Simplified domain names,<br>AdGuard French,<br>AdGuard Base,<br>AdGuard Tracking Protection,<br>AdGuard Social Media,<br>AdGuard Experimental,<br>AdGuard Spanish/Portuguese
Other extensions used: | none

### PR DESCRIPTION
**Pull request by tvinzz**
https://github.com/AdguardTeam/AdguardFilters/issues/38418

### CODE REVIEW

File: EnglishFilter/sections/antiadblock.txt [hunk 1]
```diff
@@ -45,16 +45,16 @@ hackinblood.com,alexgoldcheidt.com,pleximum.com,arcadepunks.com,sysnettechsoluti
...
...,smsbomber.in,smscodeonline.com,smskola.com,smslisten.com,smsnumbersonline.com,smsreceiveonline.com,smsreceiving.com,smstibo.com,smtebooks.net,snowfiles.com,sobootlogo.blogspot.com,sofiotheque.info,solowarez.org,somosbinarios.es,sorcerers.net,speed4up.net,spiratedhub.com,sportbh.stream,sportbikerider.us,sportsvideo.net,startseite.to,steamid.eu,steamid.uk,stefaniescott.org,stfly.io,stream4free.live,streamflix.ru,streamingdivx.co,streamlord.com,streamplay.to,studychacha.com,studydhaba.com,stupdroid.com,sukkisukki.com,supercineonline.tv,superiorwallpapers.com,superyachtfan.com,swiggygold.com,syriancivilwarmap.com,sznpaste.net,talkittechy.com,tamiltunes.online,tasksshare.com,techfinda.com,techmazza.com,technocanvas.xyz,technoslek.com,techposts.org,techreviewpro.com,techtremendous.com,techtunes.co,techtunes.com.bd,techzac.com,tecknity.com,teqniaplus.com,teqnwy.com,tercihiniyap.net,teutorrent.com,textmechanic.com,thebeast1.com,thefirmwareclub.com,themefire.net,themeslide.com,themesradar.com,therevcounter.co.uk,theseotools.net,thewpclub.net,thoptv.cricket,thoptv.net,thuthuatjb.com,tibiabr.com,timeforbitco.in,tinypaste.cc,tohax.com,tokyoghoulre.net,tomato.to,tooyul.co,topvideosgay.com,topz
```
Comment (AdamWr) [inline EnglishFilter/sections/antiadblock.txt]: Unfortunately it doesn't work on my end.
There is no `eval`.
<details><summary>Screenshot</summary>

image

</details>

So, I think that in this case we should add new rule, for example:
`imaginetricks.com$$script[tag-content="getElementById('babasbmsgx')"][min-length="5000"][max-length="20000"]`
or just
`imaginetricks.com$$script[tag-content="babasbmsgx"][min-length="5000"][max-length="20000"]`

It seems that `#%#//scriptlet("prevent-bab")` also doesn't work in this case.
This rule should fix it:
`imaginetricks.com#%#AG_abortInlineScript(/getElementById\('babasbmsgx'\)/, 'document.addEventListener');`

**Review by AdamWr — CHANGES_REQUESTED** [review]

File: EnglishFilter/sections/antiadblock.txt (hunk 1 above)
Comment (Alex-302) [reply EnglishFilter/sections/antiadblock.txt]: please add new rules with a comment, because this script is different

Comment (tvinzz) [reply EnglishFilter/sections/antiadblock.txt]: @AdamWr, your rules work perfect, thank you.

**Review by AdamWr — APPROVED** [review]

File: EnglishFilter/sections/antiadblock.txt [hunk 2]
```diff
@@ -58,6 +58,9 @@ dl2f.com,adnet.cash,warscrap.io,kimchi.tv,nulledpremium.com,freephonenum.com,rea
...
+! Different BlockAdBlock script
+imaginetricks.com$$script[tag-content="babasbmsgx"][min-length="5000"][max-length="20000"]
+imaginetricks.com#%#AG_abortInlineScript(/getElementById\('babasbmsgx'\)/, 'document.addEventListener');
```
Comment (Alex-302) [inline EnglishFilter/sections/antiadblock.txt]: better use scriptlet. JS rule will not be loaded in Opera and FF

**Review by Alex-302 — CHANGES_REQUESTED** [review]

**Review by Alex-302 — APPROVED** [review]

**Review by Alex-302 — APPROVED** [review]

**Pull request merged**